nmmo/task/gold_predicate.py

```diff
@@ -0,0 +1,88 @@
+# TODO: the below line will be gone after implementation
+# pylint: disable=unnecessary-pass
+from nmmo.task.predicate import Predicate
+
+class GoldPredicate(Predicate):
+  def __init__(self, amount: int):
+    super().__init__(amount)
+    self.amount = amount
+
+
+class HoardGold(GoldPredicate):
+  def __call__(self, team_gs, ent_id):
+    """True if the gold of agent (ent_id) is greater than or equal to min_amount.
+       Otherwise false.
+    """
+    super().__call__(team_gs, ent_id)
+    agent = team_gs.entity_or_none(ent_id)
+    if agent:
+      return agent.gold >= self.amount
+
+    return False
+
+
+class TeamHoardGold(GoldPredicate):
```

Comment:
Why can't these predicates take a TaskTarget and work on individuals or groups? Why do we want TeamHoardGold in addition to HoardGold?

Reply:
I'm posting the code snippet that we discussed below. We aim to remove Team predicates with this new design.

Reply:
All predicates are modified to work with a list of agents, so all the previous team predicates are removed.